spec/collator.html

```diff
@@ -73,23 +76,22 @@
           1. If the string given in the Type column of the row is *"boolean"* and value is not *undefined*, then
             1. Let _value_ be ! ToString(_value_).
           1. Set _opt_.[[&lt;_key_&gt;]] to _value_.
-        1. Let _relevantExtensionKeys_ be %Collator%.[[relevantExtensionKeys]].
-        1. Let _r_ be ResolveLocale(%Collator%.[[availableLocales]], _requestedLocales_, _opt_, _relevantExtensionKeys_, _localeData_).
-        1. Set _collator_.[[locale]] to _r_.[[locale]].
+        1. Let _relevantExtensionKeys_ be %Collator%.[[RelevantExtensionKeys]].
+        1. Let _r_ be ResolveLocale(%Collator%.[[AvailableLocales]], _requestedLocales_, _opt_, _relevantExtensionKeys_, _localeData_).
+        1. Set _collator_.[[Locale]] to _r_.[[locale]].
```

Comment:
the problem with ResolveLocale is that it creates an arbitrary record with fields, and we use that to match the keys provided via `options`. We could attempt to do the same, but those are not internal slots, just fields in a record. I think we can keep them like that. I know it is confusing though.